src/main/java/org/bananarama/crud/DeleteOperation.java

```
@@ -53,4 +53,11 @@
      * @return
      */
     public DeleteOperation<T> from(Stream<T> data,QueryOptions options);
+    
+    /**
+     * Delete all records from the persisted layer.
+     * @param data The collection of objects to be removed
+     * @return
+     */
+    public DeleteOperation<T> all();
```

I suggest to declare this method as `default`, with a no-op implementation, maybe with a message logged. This would avoid all the conflicts with the existing adapters that do are not implementing the `all()` method.

Reference for [default methods](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html)

---

:confused: I don't know. I want to break the compatibility with the other adapters.
But if we want to guarantee the backward compatibility, I would prefer to throw an exception:

`    default public DeleteOperation<T> all(){
        throw new UnsupportedOperationException("Not supported yet.");
    }`

---

:+1:  for throwing an exception

@cr0wbar the last words are yours :smile:

---

Ok with breaking backwards compatibility (this is an API extension and as such should break existing Adapter implementation but not client code). There must be a similar method with QueryOptions arg for consistency with similar methods for reading data. @totomz let's avoid default implementations that throw UnsupportedOperationException.